crates/cast/src/lib.rs

```diff
@@ -95,12 +98,69 @@ pub fn decode_chain_id(chain_id: FieldElement) -> String {
     String::from_utf8(non_zero_bytes).unwrap_or_default()
 }
 
-pub fn get_account<'a>(
+pub async fn get_account<'a>(
+    account: &str,
+    accounts_file: &Utf8PathBuf,
+    keystore: &Option<Utf8PathBuf>,
+    provider: &'a JsonRpcClient<HttpTransport>,
+) -> Result<SingleOwnerAccount<&'a JsonRpcClient<HttpTransport>, LocalWallet>> {
+    let chain_id = get_chain_id(provider).await?;
+    let account = match keystore {
+        Some(keystore) => get_account_from_keystore(provider, chain_id, keystore, account)?,
+        None => get_account_from_accounts_file(account, accounts_file, provider, chain_id)?,
+    };
+
+    Ok(account)
+}
+
+fn get_account_from_keystore<'a>(
+    provider: &'a JsonRpcClient<HttpTransport>,
+    chain_id: FieldElement,
+    keystore_path: &Utf8PathBuf,
+    account: &str,
+) -> Result<SingleOwnerAccount<&'a JsonRpcClient<HttpTransport>, LocalWallet>> {
+    if !keystore_path.exists() {
+        bail!("keystore file does not exist");
+    }
+    if account.is_empty() {
+        bail!("--account is required when using keystore");
```

Comment:
Maybe something like?

```suggestion
        bail!("Path passed with --account cannot be empty!");
```